### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by shanmugamlm**

### CODE REVIEW

File: src/js/contentscript.js [hunk 1]
```diff
@@ -66,3 +94,10 @@ function IsTopWindow() {
     return false;
   }
 }
+
+
+function isNotChromeURL() {
```
Comment (rajasekar-ideas2it) [inline src/js/contentscript.js]: check if this method is used

File: src/js/popup.js [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-  chrome.tabs.query({ active: true, currentWindow: true }, function(tabs) {
+  chrome.tabs.query({ active: true, currentWindow: true }, function (tabs) {
     const { id: currentTabId } = tabs[0];
-    chrome.extension.getBackgroundPage().startRecording(currentTabId);
+    chrome.extension.getBackgroundPage().initiateScreenShot(currentTabId);
   });
 };
 
-const stopRecordingBtn = document.getElementById("stopRecordingBtn");
+// startRecordingBtn.onclick = () => {
```
Comment (rajasekar-ideas2it) [inline src/js/popup.js]: remove commented code

File: src/js/contentscript.js [hunk 3]
```diff
@@ -25,33 +32,54 @@ chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
...
+
+  let screenShots = JSON.parse(localStorage.getItem('screen_shots'))
+  if (screenShots == undefined)
+    screenShots = []
+  screenShots.push(payload)
+  localStorage.setItem('screen_shots', JSON.stringify(screenShots))
+}
+
 function AddConsolejs() {
+  console.log("entried in add console11");
```
Comment (rajasekar-ideas2it) [inline src/js/contentscript.js]: remove console logs

File: src/js/contentscript.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 };
 console.log("Content script screen recorder");
-
-chrome.runtime.onMessage.addListener(function(request, sender, sendResponse) {
+// localStorage.setItem('consoleLogs','[]')
+// if(isNotChromeURL())
+chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
   switch (request.action) {
+    case constants.SCREENSHOT:
+      addScreenShots(request.payload)
```
Comment (rajasekar-ideas2it) [inline src/js/contentscript.js:11]: check if required

File: src/js/console.js [hunk 5]
```diff
@@ -23,33 +31,98 @@ function formattedMessages(messages) {
     return message;
   });
   return formattedMessages;
+}; */
+function formattedMessages(message) {
+  if (typeof message === 'string') {
+    return message;
+  }
```
Comment (rajasekar-ideas2it) [inline src/js/console.js:38]: change the order and shift the stack to the top since stack trace recording is important

File: notes.txt [hunk 6]
```diff
@@ -0,0 +1,17 @@
+shooter
+
+1.could not create folder via browser js
+2.                  zip with files
+    we can sent to server for zip 
+3.mail could not send with client browser////
+    we can do it with server also
+
+to do
```
Comment (rajasekar-ideas2it) [inline notes.txt:9]: remove this file

File: node/package.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "rest",
+  "version": "1.0.0",
+  "description": "",
+  "main": "app.js",
+  "scripts": {
```
Comment (rajasekar-ideas2it) [inline node/package.json]: strip the sample UI and service to a different code base

File: src/js/background.js [hunk 8]
```diff
@@ -28,16 +26,113 @@ var loading = false;
...
+      // link.href=window.location.href
+      let title = ''
+      if (tabs != undefined & tabs[0] != undefined)
+        title = tabs[0].title
+      let screenshotTime = new Date().toString()
+      link.download = title + '_' + screenshotTime + ".png";
+
+      link.href = screenshot.content.toDataURL();
+
+      canvas.toBlob(function (blob) {
```
Comment (rajasekar-ideas2it) [inline src/js/background.js]: save image details in local storage after successful conversion of blob

File: src/js/background.js [hunk 9]
```diff
@@ -131,50 +226,160 @@ const getVideoDataUrl = async () => {
...
 function updateIcon() {
   var iconName = loading ?
-          './assets/images/cloud-upload64-' + ((iconIndex % 2)+1) + '.png' : './assets/images/bug16.png';
-    chrome.browserAction.setIcon({path: iconName});
+    './assets/images/cloud-upload64-' + ((iconIndex % 2) + 1) + '.png' : './assets/images/bug16.png';
+  chrome.browserAction.setIcon({ path: iconName });
 }
 
-async function stopRecording() {
+const getBlobDataUrl = (blob) => {
```
Comment (rajasekar-ideas2it) [inline src/js/background.js]: check if method is used

File: src/js/background.js [hunk 10]
```diff
@@ -257,23 +564,40 @@ chrome.runtime.onConnect.addListener(port => {
...
 
 function onAttach(tabId) {
-  chrome.debugger.sendCommand({
-    tabId: tabId
-  }, "Network.enable");
-  chrome.debugger.sendCommand({
-    tabId: tabId
-  }, "Network.clearBrowserCache");
-  chrome.debugger.onEvent.addListener(allEventHandler);
+  if (chrome.runtime.lastError) {
```
Comment (rajasekar-ideas2it) [inline src/js/background.js:457]: comment why this check is handled

File: src/js/background.js [hunk 11]
```diff
@@ -28,16 +26,113 @@ var loading = false;
...
+  content: document.createElement("canvas"),
+  data: '',
+
+  init: function () {
+    this.initEvents();
+  },
+
+  saveScreenshot: function (tabs) {
+    var image = new Image();
+    image.onload = function () {
```
Comment (rajasekar-ideas2it) [inline src/js/background.js]: Remove image canvas code since it is not needed

**Review by rajasekar-ideas2it — CHANGES_REQUESTED** [review]

File: src/js/contentscript.js (hunk 1 above)
Comment (shanmugamlm) [reply src/js/contentscript.js]: removed

File: src/js/popup.js (hunk 2 above)
Comment (shanmugamlm) [reply src/js/popup.js]: removed

File: src/js/contentscript.js (hunk 4 above)
Comment (shanmugamlm) [reply src/js/contentscript.js:11]: needed

File: src/js/console.js (hunk 5 above)
Comment (shanmugamlm) [reply src/js/console.js:38]: changed

File: src/js/background.js (hunk 9 above)
Comment (shanmugamlm) [reply src/js/background.js]: removed

File: src/js/background.js (hunk 11 above)
Comment (shanmugamlm) [reply src/js/background.js]: removed

File: src/js/background.js (hunk 8 above)
Comment (shanmugamlm) [reply src/js/background.js]: canvas removed

**Pull request closed without merging**